tests/conftest.py

```diff
@@ -525,3 +527,16 @@ def fn(exception=TransactionFailed, exc_text=None):
             assert exc_text in str(excinfo.value), (exc_text, excinfo.value)
 
     return fn
+
+
+@pytest.fixture(autouse=True)
+def check_evm_version_marker(request):
+    marker = request.node.get_closest_marker("requires_evm_version")
+    if marker:
+        assert len(marker.args) == 1
+        version = marker.args[0]
+        assert version in EVM_VERSIONS
+        if not version_check(version):
+            request.node.add_marker(
+                pytest.mark.xfail(reason="Wrong EVM version", raises=EvmVersionException)
```

Comment:
i'm not a huge fan of xfail here. maybe we should use `pytest.raises(EvmVersionException)` in conjunction with yield?

Reply:
I tried doing that but the test did not work. Also [tried](https://docs.pytest.org/en/7.1.x/reference/reference.html#_pytest.hookspec.pytest_runtest_call) `yield item.runtest()` with the same result.